src/views/ProfileCreation/UserName.tsx

```diff
@@ -81,28 +83,56 @@ const UserName: React.FC = () => {
   )
   const isUserCreated = existingUserState === ExistingUserState.CREATED
 
-  const checkUsernameValidity = debounce(async (value: string) => {
-    try {
-      setIsLoading(true)
-      const res = await fetch(`${API_PROFILE}/api/users/valid/${value}`)
+  const [usernameAvailability, setUsernameAvailability] = useState<string>(undefined)
+  const debouncedUsernameAvailability = useDebounce(usernameAvailability, 200)
 
-      if (res.ok) {
-        setIsValid(true)
-        setMessage('')
-      } else {
-        const data = await res.json()
+  useEffect(() => {
+    const fetchUsernameAvailability = async (abortSignal) => {
+      try {
+        setIsLoading(true)
+        if (!debouncedUsernameAvailability) {
+          setIsValid(false)
+          setMessage('')
+          fetchAbortSignal.current = null
+        } else {
+          const res = (await fetchWithTimeout(`${API_PROFILE}/api/users/valid/${debouncedUsernameAvailability}`, {
+            method: 'get',
+            signal: abortSignal,
+            timeout: 30000,
+          })) as Response
+
+          fetchAbortSignal.current = null
+
+          if (res.ok) {
+            setIsValid(true)
+            setMessage('')
+          } else {
+            const data = await res.json()
+            setIsValid(false)
+            setMessage(data?.error?.message)
+          }
+        }
+      } catch (e) {
         setIsValid(false)
-        setMessage(data?.error?.message)
+        setMessage(t('Error fetching data'))
+      } finally {
+        setIsLoading(false)
       }
-    } finally {
-      setIsLoading(false)
     }
-  }, 200)
+
+    if (fetchAbortSignal.current) {
+      fetchAbortSignal.current.abort()
```

Comment:
hm... when you `.abort()` the request that request gets rejected with AbortError. I wonder if that would make an error message appear for a short time. 🤔 
Have you tested it somehow during the request cancellation? I know it might be tricky but worth taking a look 👀

Reply:
Correct, I added a check for that